blog/_posts/2021-11-04-docker-build-alternatives.md

```diff
@@ -0,0 +1,234 @@
+well-optimized to run locally and builds a single binary. Ko builds your Go app
+locally, then packs it into an image.
+
+
+A while back, I wrote a small extension that lets you use `ko` in Tilt. Here's what it looks like:
+
+```
+load('ext://ko', 'ko_build')
+
+ko_build(
+  'example-go-image',
+  './',
+  deps=['.']
+)
+```
+
+The upside of `ko` is that it's super simple to get started.
+
+The downside is that if you want any features that can't be expressed in a
+static Go binary, you're going to have a difficult time. Not impossible, but
+difficult. This includes: static files on-disk, dynamic C libraries, or
+hot-reload.
+
+Read the full docs on the `ko` extension
+[here](https://github.com/tilt-dev/tilt-extensions/tree/master/ko).
+
+## Pack
+
+[`pack`](https://buildpacks.io/docs/tools/pack/) is also targeted at people who
+are not happy maintaining Dockerfiles.
+
+But where Ko takes the viewpoint that you don't need a container to build an
+image, Buildpack takes the viewpoint that you need MORE containers.
+
+Wait, wait, don't run away. Let me explain.
+
+With `pack`, there's a stronger separation between the container you use
+to build an image and the container you end up running. The container images you use
+to build stuff are called buildpacks.
+
+Separating these out has some big advantages!
+
+- You can stack buildpacks to mix in the features you want (like hot-reload tools).
+
+- The buildpack keeps out lots of junk you don't want in the final image (like compilers).
+
+The `pack` extension was one of the first Tilt extensions! [Gareth
+Rushgrove](https://github.com/garethr) added it when we were still experimenting
+with the extension system.
+
+Since then, we've done work with [Daniel
+Makusa](https://github.com/dmikusa-pivotal) of the buildpacks team to make it
+even better.
+
+Here's what it looks like in a Tiltfile:
+
+```python
+pack(
+  'example-python-image', 
+  '.',
+  buildpacks=[
+    'gcr.io/paketo-buildpacks/python',
+    'gcr.io/paketo-buildpacks/environment-variables',
+  ],
+  env_vars=['BPE_FLASK_ENV=development'],
+  live_update=[
+    sync('.', '/workspace'),
+  ])
+```
+
+In this example, we're using two buildpacks: a `python` buildpack that sets up a
+Python image, and a `environment-variables` buildpack that sets
+`FLASK_ENV=development` to get live-reloading (Flask is the python framework
+that does the live-reload.)
+
+Read the full docs on the `pack` extension
+[here](https://github.com/tilt-dev/tilt-extensions/tree/master/pack).
+
+### Fun aside: Future Buildpack Work
+
+The layering approach to adding new functionality is getting a lot of excitement
+right now from container nerds.
+
+For security nerds, you can better track where source files come from and attach
+SBOMs (which are way beyond the scope of this post, but are fun to say out
+load.)
+
+One of the big reasons Tilt nerds are excited about buildpacks is that there's a
```

Comment:
```suggestion
One of the big reasons Tilters are excited about buildpacks is that there's a
```
I feel like you are using the term "nerds" a lot, you might want to switch it up once in a while 🤓